contrib/pyln-client/pyln/client/gossmapstats.py

```diff
@@ -0,0 +1,214 @@
+from pyln.client import Gossmap, GossmapChannel, GossmapNode, GossmapHalfchannel, LnFeatureBits
+from typing import Iterable, List, Optional
+
+import operator
+import statistics
+
+
+class GossmapStats(object):
+    def __init__(self, g: Gossmap):
+        self.g = g
+
+    def filter_nodes_ratelimited(self, nodes: Optional[Iterable[GossmapNode]] = None) -> List[GossmapNode]:
+        """ Filters nodes being marked by cln as ratelimited, when they send out too many updates. """
+        if nodes is None:
+            nodes = self.g.nodes.values()
+        return [n for n in nodes if n.hdr is not None and n.hdr.ratelimit]
```

Comment:
I wonder if a a generic `filter()` method wouldn't be sufficient, when one of the arguments is a filtering function that is to be applied. That would allow us implement the iteration and `if nodes` logic only once, deduplicating and simplifying the code.

Reply:
Yes, I will add three generic `filter_nodes`, `filter_channels` and `filter_halfchannels` that take a lambda or filtering function... Let me try

Reply:
(And maybe reuse that in all the filters so duplicated code is gone for good)

Reply:
Done. Currently the filters all return lists of the objects which then can optionally be cast using `set()` to make operations (union, difference, etc) on these sets. If you want I can change it to `return filter(predicate, nodes)`. Then everyone has to cast it either in `list()` or `set()` instead of directly using it, as `filter()` only returns an iterator.

I leave this unresolved so we can discuss. If you want I can share you a gist that shows how to use `gossmapstats.py` e.g. to identify good peers. I'm thinking to get this into a plugin eventually.